### DISCUSSION TITLE
@close and @after-leave events on base modal component

### DISCUSSION THREAD

**Discussion by Cannonb4ll** [inertiaui/modal#38] [Ideas]
As of now, you can add these events on the ModalLink component. But I'd love to add it to my main modal component:

```html
<Modal @close="myCloseHandler></Modal>
```

Would you be interested in adding that?

Thanks!

**Comment by pascalbaljet** [comment]
Yeah, that would be nice! I'll look into it.

**Comment by pascalbaljet** [comment]
Added in v0.18: https://inertiaui.com/inertia-modal/docs/basic-usage#modal-events
